Mathlib/RingTheory/Polynomial/ScaleRoots.lean

```diff
@@ -91,6 +93,18 @@ theorem monic_scaleRoots_iff {p : R[X]} (s : R) : Monic (scaleRoots p s) ↔ Mon
   simp only [Monic, leadingCoeff, natDegree_scaleRoots, coeff_scaleRoots_natDegree]
 #align polynomial.monic_scale_roots_iff Polynomial.monic_scaleRoots_iff
 
+theorem map_scaleRoots (p : R[X]) (x : R) (f : R →+* S) (h : f p.leadingCoeff ≠ 0) :
+    (p.scaleRoots x).map f = (p.map f).scaleRoots (f x) := by
+  ext
+  simp [Polynomial.natDegree_map_of_leadingCoeff_ne_zero _ h]
+#align polynomial.map_scale_roots Polynomial.map_scaleRoots
+
+end Semiring
+
+section CommSemiring
+
+variable [Semiring S] [CommSemiring R] [CommSemiring A] [Field K]
```

Comment:
Does this generalize to non-commutative `A`? (I can check later when a cache is available)

Reply:
Yes, but not trivially, so fine for a follow-up PR:
```lean
theorem scaleRoots_eval₂_mul {p : S[X]} (f : S →+* R) (r : R) (s : S)
    (h : Commute (f s) r) :
    eval₂ f (f s * r) (scaleRoots p s) = eval₂ f r p * f s ^ p.natDegree :=
  calc
    _ = (scaleRoots p s).support.sum fun i =>
          f (coeff p i * s ^ (p.natDegree - i)) * (f s * r) ^ i :=
      by simp [eval₂_eq_sum, sum_def]
    _ = p.support.sum fun i => f (coeff p i * s ^ (p.natDegree - i)) * (f s * r) ^ i :=
      (Finset.sum_subset (support_scaleRoots_le p s) fun i _hi hi' => by
        let this : coeff p i * s ^ (p.natDegree - i) = 0 := by simpa using hi'
        simp [this])
    _ = p.support.sum fun i : ℕ => f (p.coeff i) * f s ^ (p.natDegree - i + i) * r ^ i :=
      (Finset.sum_congr rfl fun i _hi => by
        simp_rw [f.map_mul, f.map_pow, pow_add, h.mul_pow, mul_assoc])
    _ = p.support.sum fun i : ℕ => f (p.coeff i) * (r ^ i) * f s ^ p.natDegree :=
      (Finset.sum_congr rfl fun i hi => by
        rw [tsub_add_cancel_of_le, ((h.pow_left _).pow_right _).right_comm]
        exact le_natDegree_of_ne_zero (Polynomial.mem_support_iff.mp hi))
    _ = p.support.sum (fun i : ℕ => f (p.coeff i) * r ^ i) * f s ^ p.natDegree := Finset.sum_mul.symm
    _ = eval₂ f r p * f s ^ p.natDegree := by simp [eval₂_eq_sum, sum_def]
```